test/PandoraInterface.cxx

```diff
@@ -1038,12 +1039,17 @@ void CreateSPMCParticles(const LArSPMC &larspmc, const pandora::Pandora *const p
         const std::string reaction = GetNuanceReaction((*larspmc.m_ccnc)[nuIndex], (*larspmc.m_mode)[nuIndex]);
         mcParticleParameters.m_nuanceCode = GetNuanceCode(reaction);
 
-        // Set unique parent integer address using trackID
-        const int trackID = (*larspmc.m_mcp_id)[i];
+        // Set unique parent integer address using trackID. Need to add a large enough
+        // offset of 10^6 to make these unique when we have more than 1 neutrino per event.
+        // The mcp_id's reset (to zero) per neutrino interaction vertex, and the offset
+        // should allow up to 10^6 hits for each neutrino. The true neutrino IDs (nuID)
+        // found earlier are offset by 10^8, which should allow unique trackID's for up to
+        // 100 neutrino interactions per event, each containing up to 10^6 hits
```

Comment:
What does happen if there are 101 neutrinos (or more than 10^6 hits)? Do we need a failsafe to simply drop those events/hits along with a warning, rather than risk a possible segfault?

Reply:
A check is now made to make sure the trackIDs don't go above the 100 neutrino limit; the rest of the hits will be ignored with a warning printed. A try-catch exception is also used when creating each MCParticle to make sure they contain valid, unique trackIDs and don't have NaNs.

For the non-MC hits, the position and energy are also checked. If they contain NaN values, then the hit is skipped and the next hit is processed.